Core/GDCore/Extensions/Builtin/Capabilities/AnimatableExtension.cpp

```diff
@@ -130,6 +130,30 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsAnimatableExtension(
       .AddParameter("object", _("Object"))
       .AddParameter("behavior", _("Behavior"), "AnimatableBehavior")
       .MarkAsSimple();
+
+  aut.AddExpressionAndConditionAndAction(
+         "number",
+         "ElapsedTime",
+         _("Animation elapsed time"),
+         _("the elapsed time from the beginning of the animation (in seconds)"),
+         _("the animation elapsed time"),
+         _("Animations and images"),
+         "res/actions/animation24.png")
+      .AddParameter("object", _("Object"))
+      .AddParameter("behavior", _("Behavior"), "AnimatableBehavior")
+      .UseStandardParameters(
+          "number", gd::ParameterOptions::MakeNewOptions().SetDescription(
+                        _("Elapsed time (in seconds)")))
+      .MarkAsAdvanced();
+
+  aut.AddExpression(
+         "Duration",
+         _("Animation duration"),
+         _("Return the animation duration (in seconds)."),
+         _("Animations and images"),
+         "res/actions/animation24.png")
+      .AddParameter("object", _("Object"))
+      .AddParameter("behavior", _("Behavior"), "AnimatableBehavior");
```

Comment:
Isn't the animation name missing?

Reply:
Some methods in the runtime use it, some don't. I think it should be clarified

Reply:
It's the current animation duration.
Event-based extensions won't necessarily know the animation name so it seems easier this way.
I'll clarify it in the description.